src/main/java/com/github/speedwing/log4j/cloudwatch/appender/CloudwatchAppender.java

```diff
@@ -85,54 +90,60 @@ public void setMessagesBatchSize(int messagesBatchSize) {
 
     @Override
     protected void append(LoggingEvent event) {
+        // Make sure that we don't get stuck in an infinite loop
+        if (event.getLocationInformation().getClassName().startsWith("com.amazonaws")) {
+            return;
+        }
+    
         if (cloudwatchAppenderInitialised.get()) {
             loggingEventsQueue.offer(event);
         } else {
             // just do nothing
         }
     }
 
-    private synchronized void sendMessages() {
-        LoggingEvent polledLoggingEvent;
-
-        List<LoggingEvent> loggingEvents = new ArrayList<>();
+    private void sendMessages() {
+        synchronized(sendMessagesLock) {
```

Comment:
how does this differ from synchronized at method level? I haven't done java in long long long time....